src/shared-components/alert/index.tsx

```diff
@@ -78,36 +96,34 @@ class Alert extends React.Component {
       return;
     }
 
-    // eslint-disable-next-line no-undef
     this.timer = window.setTimeout(
       this.alertExitHandler,
       Number(duration) * 1000 - ANIMATION_DELAY,
     );
   }
 
-  componentWillUnmount() {
+  componentWillUnmount(): void {
     if (this.timer) {
-      // eslint-disable-next-line no-undef
       window.clearTimeout(this.timer);
     }
   }
 
-  alertExitHandler = () => {
+  alertExitHandler = (): void => {
     const { onExit, ...rest } = this.props;
     this.setState({ exiting: true });
 
-    // eslint-disable-next-line no-undef
     window.clearTimeout(this.timer);
 
-    // eslint-disable-next-line no-undef
     window.setTimeout(() => {
       this.setState({ exited: true });
 
-      onExit({ ...rest });
+      if (onExit) {
+        onExit({ ...rest });
+      }
     }, ANIMATION_DELAY);
   };
 
-  render() {
+  render(): JSX.Element | null {
```

Comment:
My personal preference, in most cases, is the only define the return type if it is unexpected, like this. Did you define the `void` return types above in other lifecycle methods because of eslint errors, warnings, or just a preference for completeness?

Reply:
eslint warning. I see in Curology it is turned off. do you think I can apply that rule here as well?
`  "@typescript-eslint/explicit-function-return-type": "off",`

Reply:
@daigof Yeah we can turn it off 🙂  I think there's a time it's helpful but all the yellow highlight and terminal noise isn't all that helpful.

Reply:
will apply same rule as curology for consistency